iscsi/get.go

```diff
@@ -0,0 +1,311 @@
+					istats.Tpgt[i].Luns = append(istats.Tpgt[i].Luns, lun)
+				}
+			}
+		}
+	}
+	return &istats, nil
+}
+
+// isPathEnable is a utility function
+// check if the file "enable" contain enable message
+func isPathEnable(path string) (bool, error) {
+	var isEnable bool
+	isEnable = false
+	enableReadout, err := ioutil.ReadFile(filepath.Join(path, "enable"))
+	if err != nil {
+		return false, fmt.Errorf("iscsi: isPathEnable error %v", err)
+	}
+	enableReadoutNum, err := strconv.Atoi(strings.TrimSpace(string(enableReadout)))
+	if err != nil {
+		return false, fmt.Errorf("Atoi fail %v", err)
+	}
+	if enableReadoutNum > 0 {
+		isEnable = true
+	}
+	return isEnable, nil
+}
+
+func getLun(tpgtPath string) (matches []string, err error) {
+	matches, err = filepath.Glob(filepath.Join(tpgtPath, "lun/lun*"))
+	if err != nil {
+		return nil, fmt.Errorf("getLun error  %v", err)
+	}
+	return matches, nil
+}
+
+func getLunLinkTarget(lunPath string) (lunObject LUN, err error) {
+	lunObject.Name = filepath.Base(lunPath)
+	lunObject.LunPath = lunPath
+	files, err := ioutil.ReadDir(lunPath)
+	if err != nil {
+		return lunObject, fmt.Errorf("getLunLinkTarget: ReadDir path %s error %v", lunPath, err)
+	}
+	for _, file := range files {
+		fileInfo, _ := os.Lstat(lunPath + "/" + file.Name())
+		if fileInfo.Mode()&os.ModeSymlink != 0 {
+			target, err := os.Readlink(lunPath + "/" + fileInfo.Name())
+			if err != nil {
+				return lunObject, fmt.Errorf("getLunLinkTarget: Readlink err %v", err)
+			}
+			targetPath, objectName := filepath.Split(target)
+			_, typeWithNumber := filepath.Split(filepath.Clean(targetPath))
+
+			realUdev, err := ioutil.ReadFile(lunPath + "/" + target + "/udev_path")
+			if err != nil {
+				return lunObject, fmt.Errorf("getLunLinkTarget: udev_path read file %s, err %v", lunPath+"/"+target+"/udev_path", err)
+			}
+			lunObject.ObjectName = objectName
+
+			udevNumber := strings.TrimSpace(string(realUdev))
+			re := regexp.MustCompile("[0-9]+$")
+			typeNumber := re.FindString(udevNumber)
+			if typeNumber == "" {
+				typeNumber = "0"
+			}
+
+			splitTypeWithNumber := strings.Split(typeWithNumber, "_")
+			backstoreType := splitTypeWithNumber[0]
+			if len(splitTypeWithNumber) == 3 {
+				backstoreType = fmt.Sprintf("%s_%s", splitTypeWithNumber[0], splitTypeWithNumber[1])
+				typeNumber = fmt.Sprintf("%s", splitTypeWithNumber[2])
+			}
+			lunObject.Backstore = backstoreType
+			lunObject.TypeNumber = typeNumber
+			return lunObject, nil
+		}
+	}
+	return lunObject, errors.New("iscsi: getLunLinkTarget: Lun Link does not exist")
+}
+
+// ReadWriteOPS read and return the stat of read and write in megabytes,
+// and total commands that send to the target
+func ReadWriteOPS(iqnPath string, tpgt string, lun string) (readmb uint64,
+	writemb uint64, iops uint64, err error) {
+
+	readmbPath := filepath.Join(iqnPath, tpgt, "lun", lun,
+		"statistics/scsi_tgt_port/read_mbytes")
+
+	if _, err := os.Stat(readmbPath); os.IsNotExist(err) {
```

Comment:
This check seems a bit unnecessary since readUintFromFile will provide a similar error if the file doesn't exist.  Maybe this can be removed?  Or is there a reason to check specifically for file existence before trying to read the file?

Reply:
I think I'm just trying to clear out the error type, before the iqn setup and enable all those status may not exist. Compare to if the read out is something unreadable that should be a different reason to debug from the user that using the lio module.

Reply:
The error message returned from ioutil.ReadFile if the file is missing will say something like "`open <filename>: No such file or directory`", which is a fairly common error message, so I think it would be clear enough to the user.

Reply:
Thanks for explaining it. I removed it and pushed.